### PR TITLE
Qt conan

### PR DESCRIPTION
**Pull request by chhitz**
Use conan to install qt

### CODE REVIEW

**Review by meshell — COMMENTED** [review]
See below.

File: apps/qt/CMakeLists.txt [hunk 1]
```diff
@@ -13,25 +13,24 @@ set(CMAKE_AUTORCC ON)
...
-    ${CMAKE_CURRENT_LIST_DIR}/main.cpp ${CMAKE_CURRENT_LIST_DIR}/mainwindow.cpp
+    ${CMAKE_CURRENT_LIST_DIR}/main.cpp 
+    ${CMAKE_CURRENT_LIST_DIR}/mainwindow.cpp
     ${CMAKE_CURRENT_LIST_DIR}/mainwindow.h
     ${CMAKE_CURRENT_LIST_DIR}/mainwindow.ui
 )
 
 add_executable(coronan::App ALIAS ${PROJECT_NAME})
 
-find_package(
```
Comment (meshell) [inline apps/qt/CMakeLists.txt:23]: This is according to the Qt documentation (https://doc.qt.io/qt-5/cmake-get-started.html#build-a-gui-executable). 
Is there a reason for the change?

File: external/qt_charts_conan/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+from conans import ConanFile, tools
```
Comment (meshell) [inline external/qt_charts_conan/conanfile.py]: Why is this file ? conan-qt by bincrafters has the option qt:qtcharts to include qtcharts.

File: build.sh [hunk 3]
```diff
@@ -70,6 +70,7 @@ fi
 
 [[ -d "${BUILD_DIR}" ]] || mkdir ${BUILD_DIR}
 
+${CONAN} remote add bincrafters https://api.bintray.com/conan/bincrafters/public-conan || true
 (cd external/qt_charts_conan && ${CONAN} create .)
```
Comment (meshell) [inline build.sh]: This IMHO not required, because it is already done in the CMakeLists.txt (above).

File: apps/cli/CMakeLists.txt [hunk 4]
```diff
@@ -62,6 +62,11 @@ target_compile_options(
                            $<$<BOOL:UNIX>:-fPIC;>
 )
 
+set_target_properties(${PROJECT_NAME} PROPERTIES
```
Comment (meshell) [inline apps/cli/CMakeLists.txt]: IMHO target_compile_features PUBLIC for the library target should do the job.

**Review by meshell — COMMENTED** [review]
Why is this change required?
IMHO target_compile_features PUBLIC for the library target should do the job.

File: CMakeLists.txt [hunk 5]
```diff
@@ -19,7 +19,15 @@ include(clang)
 
 include(conan)
 
-conan_cmake_run(CONANFILE conanfile.txt BASIC_SETUP CMAKE_TARGETS BUILD poco missing)
+conan_cmake_run(REQUIRES doctest/2.3.7
```
Comment (meshell) [inline CMakeLists.txt:25]: Is there a reason not to use BASIC_SETUP ?

Comment (chhitz) [reply CMakeLists.txt:25]: actually required now (for qt)

File: external/qt_charts_conan/conanfile.py (hunk 2 above)
Comment (chhitz) [reply external/qt_charts_conan/conanfile.py]: removed

**Review by meshell — APPROVED** [review]

**Pull request closed without merging**